docs/NFF-0.1-spec.norg

```diff
@@ -0,0 +1,875 @@
+    draw parallels here+
+
+    @code norg
+    == Function definition
+    ==
+    @end
+
+** Detached Modifier Extensions
+   {# Detached modifiers} support extensions which must immediately follow the detached modifier
+   (or another extension).
+
+*** TODO Status Extension
+    The TODO item extension assigns a task status to a certain modifier. You probably know this
+    concept from Markdown where unordered lists can become tasks. In _Norg_ we take this concept to
+    the next level because any detached modifier can be assigned a task status. This can for
+    example be useful for the author of a document to keep track of the status of certain sections.
+    The syntax is the following:
+    - A todo status extension starts with a `|` char
+    - Immediately a single status indicator character must follow. The following statuses are
+      supported:
+   -- ` `: undone (a literal space)
+   -- `x`: done
+   -- `?`: needs further input/clarification
+   -- `!`: urgent
+   -- `+`: recurring
+   -- `-`: in-progress/pending
+   -- `=`: on hold
+   -- `_`: put down/cancelled
+   - Finally another `|` char closes the todo status extension
+
+*** Indent Segment Extension
+    Since {# nestable detached modifiers} can only contain a {# paragraph} this can cause severe
+    limitations because insertion of (for example) code blocks is not possible. To alleviate this
+    deficiency, the {# indent segment extension} exists, which increases the indentation of the
+    tree by one level allowing nesting of more complex elements to occur.
+    The syntax is the following:
+    - The indent segment extension starts with a single `\` char
+    - This must be _immediately_ followed by a {# line ending}
+    -- Thus, the start node of this extension can be seen as a {# trailing modifier}
+    - The contents follow
+    - The extension ends when:
+    -- Another {# nestable detached modifier} occurs
+    -- Or when the indentation is manually closed with one of the {# delimiting modifiers}
+     +TODO(vhyrro): discuss what truly "automatically" ends the indent segment. I propose that the only way
+     to auto-close an indent segment is to use a structural detached modifier or a modifier of the same
+     type as the parent of the indent segment; i.e. if you attach an indent segment to a level 1
+     unordered list item then creating another level 1 unordered list item auto-closes the indent
+     segment. However, creating a level 1 ordered list segment inside would be okay.+
+
+    @code norg
+    - \
+      The indent segment extension has started.
+
+      I can now include multiple paragraphs within this list item.
+
+      @code lua
+      print("Even code blocks are allowed!")
+      @end
+
+      ^ Range-able detached modifiers can be inserted, too.
+
+      ---
+    - Another nestable detached modifier ends the indent segment
+    - However, in this example I already closed the indent segment manually with `---`
+    @end
```

Comment:
This syntax currently does not work because the block ends prematurely when encountering the `@end` of the internal `@code lua` block. We need to switch to an `#example` tag instead (this holds practically for all `@code norg` blocks in this document).